src/runtime-rs/crates/hypervisor/src/qemu/inner.rs

```diff
@@ -167,10 +167,19 @@ impl QemuInner {
     pub(crate) async fn stop_vm(&mut self) -> Result<()> {
         info!(sl!(), "Stopping QEMU VM");
         if let Some(ref mut qemu_process) = &mut self.qemu_process {
-            info!(sl!(), "QemuInner::stop_vm(): kill()'ing qemu");
-            qemu_process.kill().await.map_err(anyhow::Error::from)
+            let is_qemu_running = qemu_process.id().is_some();
+            if is_qemu_running {
+                info!(sl!(), "QemuInner::stop_vm(): kill()'ing qemu");
+                qemu_process.kill().await.map_err(anyhow::Error::from)
+            } else {
+                info!(
```

Comment:
Nit: warn!() or info!(), which one is better ?

Reply:
That's a good question... I stayed with `info!()` in the end on the base that this PR makes the qemu driver accept multiple `stop_vm()` calls as normal, business as usual, so from that standpoint there's kind of nothing to warn about. :-)

Reply:
Warnings are usually devoted to something that we want the user to be aware of so that they can _maybe_ take action. In this case, it seems that the warning would popup every time and the user cannot really do anything about it. I'd keep `info!()`.